src/pkg/scheduler/mock.go

```diff
@@ -0,0 +1,17 @@
+// Copyright Project Harbor Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package scheduler
```

Comment:
Please move the mock object to the same packages under src/testings

Reply:
The mocked DAO interface here is needed only by the scheduler, and the `src/testing` directory contains the mocks shared with others

Reply:
If it is only DAO mock objects, I think it's ok.

But the src/testing has become mapping shadow packages of keeping relevant mock objects of the interfaces located the src/(controller/pkg). Even DAO not shared can also be put there for better organization and clean code.

I suggest you generate a scheduler interface mock object in the corresponding pkg of src/testing for other components that use the Scheduler to write their test suite.